src/main/java/org/jabref/logic/importer/fetcher/CrossRef.java

```diff
@@ -137,6 +137,10 @@ private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
                             .map(year -> Integer.toString(year)).orElse("")
             );
             entry.setField(StandardField.DOI, item.getString("DOI"));
+            entry.setField(StandardField.JOURNAL, item.getString("container-title"));
+            entry.setField(StandardField.PUBLISHER, item.getString("publisher"));
+            entry.setField(StandardField.ISSUE, item.getString("issue"));
```

Comment:
```suggestion
            entry.setField(StandardField.NUMBER, item.getString("issue"));
```

Issue in biblatex is for "Spring issue" or similar.

Reply:
<html><body>
<!--StartFragment-->

BibTeX | BibLaTeX | EntryPreview/CSL | proposed logic, conditions and info
-- | -- | -- | --
volume | volume | volume |  
number | issue | issue | For conversion to CSL or BibTeX: BibLaTeX number takes priority and supersedes BibLaTeX issue
number | number | issue | For conversion to CSL or BibTeX: BibLaTeX number takes priority and supersedes BibLaTeX issue
pages | eid | number | Some journals put the article-number (= eid) into the pages field. If BibLaTeX eid exists, provide csl number to the style. If pages exists, provide csl page.  If eid WITHIN the pages field exists, detect the eid and provide csl number. If both eid and pages exists, ideally provide both csl number and csl page. Ideally the citationstyle should be able to flexibly choose the rendering. If this is not possible then map biblatex eid to CSL page, since this is sufficient to comply with APA 7th edition, but will break other citationstyles (e.g. IEEE)
pages | pages | page | Some journals put the article-number (= eid) into the pages field. If BibLaTeX eid exists, provide csl number to the style. If pages exists, provide csl page.  If eid WITHIN the pages field exists, detect the eid and provide csl number. If both eid and pages exists, ideally provide both csl number and csl page. Ideally the citationstyle should be able to flexibly choose the rendering. If this is not possible then map biblatex eid to CSL page, since this is sufficient to comply with APA 7th edition, but will break other citationstyles (e.g. IEEE)

<!--EndFragment-->
</body>
</html>


As a general reminder about the difference between number and issue, the biblatex documentation (https://ctan.kako-dev.de/macros/latex/contrib/biblatex/doc/biblatex.pdf) on page 22 shows this:

```
issue field (literal)
The issue of a journal. This field is intended for journals whose individual issues
are identified by a designation such as ‘Spring’ or ‘Summer’ rather than the month
or a number. The placement of issue is similar to month and number. Integer
ranges and short designators are better written to the number field. See also month,
number and §§ 2.3.10 and 2.3.11.
```

and on page 23:

```
number field (literal)
The number of a journal or the volume/number of a book in a series. See also
issue as well as §§ 2.3.7, 2.3.10, 2.3.11. With @patent entries, this is the number or
record token of a patent or patent request. Normally this field will be an integer or an
integer range, but it may also be a short designator that is not entirely numeric such
as “S1”, “Suppl. 2”, “3es”. In these cases the output should be scrutinised carefully.
Since number is—maybe counterintuitively given its name—a literal field, sorting
templates will not treat its contents as integers, but as literal strings, which means
that “11” may sort between “1” and “2”. If integer sorting is desired, the field can be
declared an integer field in a custom data model (see § 4.5.4). But then the sorting of
non-integer values is not well defined.
```

Reply:
See #8607